src/AbstractDifferentiation.jl

```diff
@@ -98,7 +101,7 @@ function value_and_hessian(ab::AbstractBackend, f, xs...)
             value = primalvalue(ab, v, f, xs)
             primalcalled = true
         end
-        return g
+        return g[1] # gradient returns a tuple
```

Comment:
same comment in this function

Reply:
For the `FDMBackend`s, there was no issue, e.g.,
```julia
fhess = x -> fgrad(x, yvec)
hess1 = AD.hessian(fdm_backend2, fhess, xvec)
```
works fine with a tuple as output of the gradient function. However, for ForwardDiff
```julia
struct ForwardDiffBackend2 <: AD.AbstractForwardMode end
const forwarddiff_backend2 = ForwardDiffBackend2()
AD.@primitive function pushforward_function(ab::ForwardDiffBackend2, f, xs...)
    # jvp = f'(x)*v, i.e., differentiate f(x + h*v) wrt h at 0
    return function (vs)
        if xs isa Tuple
            @assert length(xs) <= 2
            if length(xs) == 1
                (ForwardDiff.derivative(h->f(xs[1]+h*vs[1]),0),)
            else
                ForwardDiff.derivative(h->f(xs[1]+h*vs[1], xs[2]+h*vs[2]),0)
            end
        else
            ForwardDiff.derivative(h->f(xs+h*vs),0)
        end
    end
end
AD.primalvalue(::ForwardDiffBackend2, ::Any, f, xs) = ForwardDiff.value.(f(xs...))
hess1 = AD.hessian(forwarddiff_backend2, fhess, xvec)
```
an error is thrown:
```
ERROR: LoadError: MethodError: no method matching extract_derivative(::Type{ForwardDiff.Tag{var"#22#29"{Tuple{Vector{Float64}}, AbstractDifferentiation.var"#127#128"{ForwardDiffBackend2, var"#107#108"}, Tuple{Vector{Float64}}}, Int64}}, ::Tuple{Base.ReshapedArray{ForwardDiff.Dual{ForwardDiff.Tag{var"#22#29"{Tuple{Vector{Float64}}, AbstractDifferentiation.var"#127#128"{ForwardDiffBackend2, var"#107#108"}, Tuple{Vector{Float64}}}, Int64}, Float64, 1}, 1, Adjoint{ForwardDiff.Dual{ForwardDiff.Tag{var"#22#29"{Tuple{Vector{Float64}}, AbstractDifferentiation.var"#127#128"{ForwardDiffBackend2, var"#107#108"}, Tuple{Vector{Float64}}}, Int64}, Float64, 1}, Matrix{ForwardDiff.Dual{ForwardDiff.Tag{var"#22#29"{Tuple{Vector{Float64}}, AbstractDifferentiation.var"#127#128"{ForwardDiffBackend2, var"#107#108"}, Tuple{Vector{Float64}}}, Int64}, Float64, 1}}}, Tuple{Base.MultiplicativeInverses.SignedMultiplicativeInverse{Int64}}}})
Closest candidates are:
  extract_derivative(::Type{T}, ::ForwardDiff.Dual) where T at /Users/frank/.julia/packages/ForwardDiff/QOqCN/src/derivative.jl:81
  extract_derivative(::Type{T}, ::Real) where T at /Users/frank/.julia/packages/ForwardDiff/QOqCN/src/derivative.jl:82
  extract_derivative(::Type{T}, ::AbstractArray) where T at /Users/frank/.julia/packages/ForwardDiff/QOqCN/src/derivative.jl:83
Stacktrace:
```
since all options were working when vectors are returned, I included those g[1].